dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1981,35 +1907,223 @@ public Contentlet save(Contentlet contentlet) throws DotDataException, DotStateE
+            } catch (DotSecurityException e) {
+
+            }
         }
 
-        if (InodeUtils.isSet(contentlet.getFolder())) {
-            content.setFolder(contentlet.getFolder());
+        upsertValues.add(name);
+        upsertValues.add(new Timestamp(contentlet.getModDate().getTime()));
+        upsertValues.add(contentlet.getModUser());
+        upsertValues.add(new Long(contentlet.getSortOrder()).intValue());
+
+        if (allowTitle) { // if the title was not intentionally set to null.
+            upsertValues.add(name); //friendly name
+        } else {
+            upsertValues.add(null);
         }
 
-        contentletCache.remove(content.getInode());
-        HibernateUtil.evict(content);
+        upsertValues.add(contentlet.getContentTypeId());
 
-        return content;
-	}
+        addWysiwygParam(contentlet, upsertValues);
 
-	/**
+        upsertValues.add(UtilMethods.isSet(contentlet.getIdentifier())?contentlet.getIdentifier():null);
+        upsertValues.add(contentlet.getLanguageId());
+
+        final Map<String, Object> fieldsMap = getFieldsMap(contentlet);
+
+        try {
+            addDynamicFields(upsertValues, fieldsMap,"date");
+            addDynamicFields(upsertValues, fieldsMap,"text");
+            addDynamicFields(upsertValues, fieldsMap,"text_area");
+            addDynamicFields(upsertValues, fieldsMap,"integer");
+            addDynamicFields(upsertValues, fieldsMap,"float");
+            addDynamicFields(upsertValues, fieldsMap,"bool");
+        } catch (JsonProcessingException e) {
+            throw new DotDataException(e);
+        }
+
+        return upsertValues;
+    }
+
+    private void addWysiwygParam(final Contentlet contentlet, final List<Object> upsertValues) {
+        final List<String> wysiwygFields = contentlet.getDisabledWysiwyg();
+        if( wysiwygFields != null && wysiwygFields.size() > 0 ) {
+            final StringBuilder wysiwyg = new StringBuilder();
+            int j = 0;
+            for(final String wysiwygField : wysiwygFields ) {
+                wysiwyg.append(wysiwygField);
+                j++;
+                if( j < wysiwygFields.size() ) wysiwyg.append(",");
+            }
+            upsertValues.add(wysiwyg.toString());
+        } else{
+            upsertValues.add(null);
+        }
+    }
+
+    /**
+     * Add dates, text, text_area and integer fields from the contentlet's fields map to the {@link DotConnect} object
+     * @param upsertValues
+     * @param fieldsMap
+     * @param prefix
+     */
+    private void addDynamicFields(final List<Object> upsertValues, final Map<String, Object> fieldsMap, final String prefix)
+            throws JsonProcessingException {
+        Object defaultValue = null;
+
+        if (prefix.equals("integer") || prefix.equals("float")){
+            defaultValue = 0;
+        } else if (prefix.equals("bool")){
+            defaultValue = Boolean.FALSE;
+        }
+
+        for (int i = 1; i <= 25; i++) {
```

Comment:
I am reading this code and I don't understand why it has to be 25 can you set up a constant with 25 and name it somehow so readers can have a better sense

Reply:
something like max_available_fields whatever makes sense

Reply:
Done